Sources/PackageModel/PackageReference.swift

```diff
@@ -29,13 +27,8 @@ public struct PackageReference: Codable {
         case remote
     }
 
-    /// Compute identity of a package given its URL.
-    public static func computeIdentity(packageURL: String) -> String {
```

Comment:
This will break clients that currently maintain mappings from packages to their own objects, such as for IDEs.  What is the new API they should use?  Should they just call `PackageIdentity(url: url)` instead?

Reply:
What about for local paths (specified by file path) and roots?  How are their identities formed in the new API?

Reply:
could PackageReference expose the identity it is initialized with and such clients can work off of that?

Reply:
I think the right thing here is probably for clients to create a `PackageIdentity` from the given URL.  The question about local paths was more of a question about whether the diffs in this PR would handle that properly.  The URL that is passed to `PackageIdentity` would need to support local paths as well as `file://` URLs (which are different, the former being a local unversioned package reference and the latter being a local Git repository).

Reply:
Yep, agreed, supporting these cases fully is what I'll be doing in a follow-up. Do you think the interface needs to change to allow this? I would think we can pass in a file URL and depending on whether there's a repo at that location or not, we would get different behavior in terms of what we use for the identity (a repo could use the fingerprint we talked about, if it is just a directory we would essentially fallback to last path component).

Reply:
I think that the API should ideally allow creation of a package identity from either an `AbsolutePath` (for a local unversioned path reference) or a `URL` (which designates either a local repository via `file://` or a remote repository using some other scheme).  What the implementation does under-the-hood is none of the client's business, but that would allow the client to express the two quite fundamentally different kinds of package reference (specifically one is versioned and the other is taken as given).

For now we can have a single string that either has or doesn't have a `file` scheme but ideally there would be a clearer API.  But that can be added later, as you suggest (I'm trying to not keep requesting changes from Mattt here).

Reply:
766ea50 creates two initializers `PackageIdentity(url: String)` and `PackageIdentity(path: AbsolutePath)` and goes through each existing call site to figure out which one should be used. (Most of the hassle came around tests, which had used hardcoded string literals.)

See https://github.com/apple/swift-package-manager/pull/3057#issuecomment-729131108.

Reply:
Great, thank you!  That sounds like the right change from a client's perspective.